### PR TITLE
Feature/code cleanup

### PR DESCRIPTION
**Pull request by martin-v**
*Before submission, please check that ...*

- [x] the code follows the [Clean Code](https://clean-code-developer.com/) guidelines.
- [x] the necessary tests are either created or updated.
- [ ] all status checks (Travis CI, SonarCloud, etc.) pass.
- [x] your commit history is cleaned up.
- [x] you updated the changelog.
- [x] you updated necessary documentation within [retest/docs](https://github.com/retest/docs).

### CODE REVIEW

File: src/test/java/de/retest/recheck/util/ReflectionUtilitiesTest.java [hunk 1]
```diff
@@ -202,7 +202,7 @@ public void getSimpleName_should_return_classname() throws Exception {
 		assertThat( getSimpleName( "de.retest.Main$1" ) ).isEqualTo( "Main$1" );
 	}
 
-	private class A {};
+	private static class A {};
 
 	private class B extends A {};
```
Comment (dkrn03) [inline src/test/java/de/retest/recheck/util/ReflectionUtilitiesTest.java]: Sollte die Klasse B dann nicht auch static sein?

File: src/main/java/de/retest/recheck/ui/descriptors/StringAttribute.java [hunk 2]
```diff
@@ -44,10 +44,10 @@ public Integer parse( final String value ) throws ParameterParseException {
 	public static final ParameterType parameterTypeBoolean = new ParameterType( "BOOLEAN" ) {
 		@Override
 		public Boolean parse( final String value ) throws ParameterParseException {
-			if ( value.equalsIgnoreCase( "true" ) ) {
```
Comment (dkrn03) [inline src/main/java/de/retest/recheck/ui/descriptors/StringAttribute.java:47]: Was genau ist hier der Unterschied, wieso können so NPEs verhindert werden?

Comment (martin-v) [reply src/main/java/de/retest/recheck/ui/descriptors/StringAttribute.java:47]: If `value` is `null` the old code throws a NPE, while `"true".equalsIgnoreCase()` is just `false`.

File: src/test/java/de/retest/recheck/util/ReflectionUtilitiesTest.java (hunk 1 above)
Comment (martin-v) [reply src/test/java/de/retest/recheck/util/ReflectionUtilitiesTest.java]: Yes, it's just slipped into another commit: https://github.com/retest/recheck/commit/a3a433132da228449c8bc1dcbb7ca0eeca0373c9#diff-ebad5a247e6ad34c4d38ff43dbb9bf0536d7dd906ee2532746deb38ed4e8cd96L207

**Review by dkrn03 — APPROVED** [review]

**Pull request merged**